tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersHelper.cs

```diff
@@ -0,0 +1,587 @@
+                    if (entry.ServiceLabels?.Any(label => NormalizeLabel(label).Equals(NormalizeLabel(serviceLabel), StringComparison.OrdinalIgnoreCase)) == true)
+                    {
+                        return entry;
+                    }
+                    // Check PRLabels for exact match
+                    else if (entry.PRLabels?.Any(label => NormalizeLabel(label).Equals(NormalizeLabel(serviceLabel), StringComparison.OrdinalIgnoreCase)) == true)
+                    {
+                        return entry;
+                    }
+                }
+
+                // If repoPath is provided, match by PathExpression (contains match for paths - this is intentional)
+                if (!string.IsNullOrEmpty(path) && ExtractDirectoryName(entry.PathExpression).Contains(ExtractDirectoryName(path), StringComparison.OrdinalIgnoreCase) == true)
+                {
+                    return entry;
+                }
+            }
+            return null;
+        }
+
+        public (CodeownersEntry entry, bool existed) CreateOrUpdateCodeownersEntry(
+            CodeownersEntry? existingEntry,
+            string path,
+            string serviceLabel,
+            List<string> serviceOwners,
+            List<string> sourceOwners,
+            bool isAdding,
+            bool isMgmtPlane)
+        {
+            if (existingEntry != null)
+            {
+                // Update existing entry with new codeowners
+                if (isAdding)
+                {
+                    existingEntry.ServiceOwners = AddUniqueOwners(existingEntry.ServiceOwners, serviceOwners);
+                    existingEntry.SourceOwners = AddUniqueOwners(existingEntry.SourceOwners, sourceOwners);
+                }
+                else
+                {
+                    existingEntry.ServiceOwners = RemoveOwners(existingEntry.ServiceOwners, serviceOwners);
+                    existingEntry.SourceOwners = RemoveOwners(existingEntry.SourceOwners, sourceOwners);
+                }
+
+                return (existingEntry, true);
+            }
+            else
+            {
+                // Create new entry
+                if (string.IsNullOrEmpty(serviceLabel) || string.IsNullOrEmpty(path))
+                {
+                    throw new Exception($"When creating a new entry, both a Service Label and Path are required. Provided: serviceLabel = '{serviceLabel}', path = '{path}'");
+                }
+
+                List<string> serviceLabels = new List<string>() { serviceLabel };
+
+                if (isMgmtPlane)
+                {
+                    serviceLabels.Add("Mgmt");
+                }
+
+                var newEntry = new CodeownersEntry()
+                {
+                    PathExpression = path ?? string.Empty,
+                    ServiceLabels = serviceLabels ?? new List<string>(),
+                    PRLabels = new List<string>() { serviceLabel} ?? new List<string>(),
```

Comment:
This null coalesce doesn't make sense. `serviceLabel` will never be null/empty and a List (even if it's empty or contains an element that's null) is not itself null... You'll never get to the right side of the `??`